src/ToDo/Business/TaskService.cs

```diff
@@ -40,4 +40,18 @@ public async Task DeleteAsync(ToDoTask task, CancellationToken ct)
 
 		_messenger.Send(new EntityMessage<ToDoTask>(EntityChange.Delete, task), task.ListId);
 	}
+
+	/// <inheritdoc />
+	public async ValueTask<IEnumerable<ToDoTask>> GetAllAsync(string displayName = "", CancellationToken ct = default)
+	{
+		Task<TaskReponseData<TaskData>> getMethod = string.IsNullOrWhiteSpace(displayName) ?
+			_client.GetAllAsync(ct) : _client.GetByFilterAsync(displayName, ct);
+
+		return ((await getMethod).Value ?? Enumerable.Empty<TaskData>())
+		.Where(data => data.ParentList?.Id is not null)
+		.Select(data =>
+		{
+			return new ToDoTask(data.ParentList?.Id ?? string.Empty, data);
+		}).ToList();
```

Comment:
Just a recommendation for the next time: As you checked for `data.ParentList?.Id is not null` on line 51, you can "forgive the null" here:

```suggestion
		return ((await getMethod).Value ?? Enumerable.Empty<TaskData>())
		.Where(data => data.ParentList?.Id is not null)
		.Select(data => new ToDoTask(data.ParentList!.Id!, data))
		.ToList();
```

It's for those cases where you handled the `null` but the compiler is not able to detect it by itself (yet)  that the `!` exists.

Reply:
It's Okay, thank you so much